sklearn/cluster/tests/test_k_means.py

```diff
@@ -1030,3 +1030,47 @@ def test_minibatch_kmeans_wrong_params(param, match):
     # are passed for the MiniBatchKMeans specific parameters
     with pytest.raises(ValueError, match=match):
         MiniBatchKMeans(**param).fit(X)
+
+
+@pytest.mark.parametrize("param, match", [
+    ({"n_local_trials": 0},
+     r"n_local_trials is set to 0 but should be an "
+     r"integer value greater than zero"),
+    ({"x_squared_norms": X[:2]},
+     r"The length of x_squared_norms .* should "
+     r"be equal to the length of n_samples")]
+)
+def test_kmeans_plusplus_wrong_params(param, match):
+    with pytest.raises(ValueError, match=match):
+        kmeans_plusplus(X, n_clusters, **param)
+
+
+@pytest.mark.parametrize("data", [X, X_csr])
+def test_kmeans_plusplus_output(data):
+    # Check for the correct number of seeds and all positive values
+    centers, indices = kmeans_plusplus(data, n_clusters)
+
+    # Check there are the correct number of indices and that all indices are
+    # positive and within the number of samples
+    assert indices.shape[0] == n_clusters
+    assert (indices >= 0).all()
+    assert (indices <= X.shape[0]).all()
+
+    # Check for the correct number of seeds and that they are bound by the data
+    assert centers.shape[0] == n_clusters
+    assert (centers.max(axis=0) <= X.max(axis=0)).all()
+    assert (centers.min(axis=0) >= X.min(axis=0)).all()
+
+    # Check that indices correspond to reported centers
+    assert_allclose(X[indices], centers)
```

Comment:
Could you please add a similarly parametrized test to check that we get the same results when we manually pass `x_squared_norms` and when we leave it to the `None` default value?